compiler/test/fail_compilation/test12228.d

```diff
@@ -1,10 +1,9 @@
 /*
 TEST_OUTPUT:
 ---
-fail_compilation/test12228.d(13): Error: undefined identifier `this`, did you mean `typeof(this)`?
-fail_compilation/test12228.d(18): Error: no property `x` for type `object.Object`
```

Comment:
This looks like a spurios error, `typeof(super)` should be `C` here, not `Object`.